### PR TITLE
Added details, examples and docs on Gulp v4

### PR DESCRIPTION
**Pull request by IEvangelist**
Fixes https://github.com/aspnet/Docs/issues/8827
Fixes https://github.com/aspnet/Docs/issues/7632

**Internal Review Pages**
* [Use Gulp](https://review.docs.microsoft.com/en-us/aspnet/core/client-side/using-gulp?view=aspnetcore-2.1&branch=pr-en-us-8843)
* [Bundle & minify](https://review.docs.microsoft.com/en-us/aspnet/core/client-side/bundling-and-minification?view=aspnetcore-2.1&branch=pr-en-us-8843)

### CODE REVIEW

File: aspnetcore/client-side/bundling-and-minification.md [hunk 1]
```diff
@@ -265,8 +265,16 @@ If Visual Studio and/or the Bundler & Minifier extension aren't available, conve
 
 Add a *package.json* file, with the following `devDependencies`, to the project root:
 
+# [Gulp 4.x](#tab/gulp4x/)
+
+[!code-json[](../client-side/bundling-and-minification/samples/BuildBundlerMinifierApp/package2.json?range=5-13)]
+
+# [Gulp 3.x](#tab/gulp3x/)
```
Comment (scottaddie) [inline aspnetcore/client-side/bundling-and-minification.md]: Please remove the Gulp 3.x content all throughout. Showing only 4.x is fine.

File: aspnetcore/client-side/using-gulp.md [hunk 2]
```diff
@@ -60,11 +60,44 @@ The above code specifies which Node modules are required. The `require` function
 
 Once the requisite modules are imported, the tasks can be specified. Here there are six tasks registered, represented by the following code:
 
+> [!NOTE]
+> Gulp 4 introduced some significant API changes. Please be aware of the version you're using and look through the [change log](https://github.com/gulpjs/gulp/blob/master/CHANGELOG.md) for more details.
+
+# [Gulp 4.x](#tab/gulp4x/)
```
Comment (scottaddie) [inline aspnetcore/client-side/using-gulp.md]: Showing only Gulp 4.x in this doc is okay too.

File: aspnetcore/client-side/bundling-and-minification.md (hunk 1 above)
Comment (IEvangelist) [reply aspnetcore/client-side/bundling-and-minification.md]: So remove the notion of tab usage here for versioning of Gulp all together? Basically, just update the existing stuff with Gulp v4 removing v3 entirely - right?

Comment (scottaddie) [reply aspnetcore/client-side/bundling-and-minification.md]: That's correct. Thank you for the help!

Comment (IEvangelist) [reply aspnetcore/client-side/bundling-and-minification.md]: Sounds good, will update soon. Thank you 🤘

**Comment by scottaddie** [comment]
Thank you so very much for your help, @IEvangelist! Your changes will go live later today.

**Pull request merged**